pkg/destroy/vsphere/vsphere.go

```diff
@@ -189,11 +189,19 @@ func deleteTag(ctx context.Context, client *rest.Client, tagID string) error {
 
 func deleteTagCategory(ctx context.Context, client *rest.Client, categoryID string) error {
 	tagManager := tags.NewManager(client)
-	category, err := tagManager.GetCategory(ctx, categoryID)
-	if err == nil {
-		err = tagManager.DeleteCategory(ctx, category)
+	ids, err := tagManager.ListCategories(ctx)
+	if err != nil {
+		return err
 	}
-	return err
+
+	for _, id := range ids {
+		category, err := tagManager.GetCategory(ctx, id)
+		if err == nil && category.Name == categoryID {
+			return tagManager.DeleteCategory(ctx, category)
+		}
+	}
+
+	return errors.Errorf("category %s not found", categoryID)
```

Comment:
I think this should not return an error. We can print a debug statement. But the destroyer should accept a missing category as acceptable.